### PR TITLE
Add update files endpoint

### PR DESCRIPTION
**Pull request by calufornia**
## Description

Adds an endpoint `/dev/files/update` to the local dev server that will be used by the Studio to save files. For now, we just send over the contents of the file in the request, but in the future we could use SSE, WS, etc. to make this more efficient. Also adds a directory traversal check to `/dev/files/get`.

## Test plan
Tested in conjunction with web changes

### CODE REVIEW

File: pkg/server/apidev/apidev.go [hunk 1]
```diff
@@ -288,6 +293,30 @@ func GetFileHandler(ctx context.Context, state *state.State, r *http.Request) (G
...
+	// Ensure the path is within the dev server root.
+	if !strings.HasPrefix(req.Path, s.Dir) {
+		return struct{}{}, errors.New("path is outside dev root")
+	}
+
+	if strings.Contains(req.Path, "..") {
+		return struct{}{}, errors.New("path may not contain directory traversal elements (`..`)")
+	}
+
+	if err := os.WriteFile(req.Path, []byte(req.Content), 0644); err != nil {
```
Comment (ameyakhare) [inline pkg/server/apidev/apidev.go:313]: we don't validate that the path exists + is in the actual airplane directory the dev server is started from. I would add that validation here

**Review by ameyakhare — CHANGES_REQUESTED** [review]
one small thing

File: pkg/server/apidev/apidev.go (hunk 1 above)
Comment (ameyakhare) [reply pkg/server/apidev/apidev.go:313]: or if its new, that its just in the dev server folder. worried about some bug overwriting a system file lol

Comment (calufornia) [reply pkg/server/apidev/apidev.go:313]: that's the validation on lines 305-311, right?

**Review by ameyakhare — APPROVED** [review]

**Pull request merged**